packages/keystone/fetch.js

```diff
@@ -0,0 +1,41 @@
+const nodeFetch = require('node-fetch')
+
+const { getExecutionContext } = require('./executionContext')
+const { getLogger } = require('./logging')
+const Mertrics = require('./metrics')
+
+const logger = getLogger('fetch')
+
+async function fetch (url, options) {
+
+    const urlObject = new URL(url)
+    const host = urlObject.hostname
+    const path = urlObject.pathname
+    const metricUrl = host.replace(/[^a-zA-Z0-9]/g, '')
+
+    const executionContext = getExecutionContext()
+    const parentReqId = executionContext.reqId
+
+    const startTime = Date.now()
+
+    try {
+        const response = await nodeFetch(url, options)
+
+        const endTime = Date.now()
+        const elapsedTime = endTime - startTime
```

Comment:
can we track `elapsedTime` not only for successful requests, but also for errors?

Reply:
Nice suggestion :)